tests/foreman/ui/test_errata.py

```diff
@@ -1085,54 +1086,71 @@ def test_positive_filter_by_environment(
     host's previous environment
 
     :id: 78110ba8-3942-46dd-8c14-bffa1dbd5195
 
     :Setup:
-        1. Make sure multiple environments are present.
+        1. Make sure multiple environments are present, one registered host.
         2. Content host's previous environments have additional errata.
+        3. Promote the Host's content view version to a new lifecycle environment.
+        4. Set the Host to use the new environment, and original content view.
 
-    :steps: Go to Content Hosts -> Select content host -> Errata Tab ->
-        Select Previous environments.
+    :Steps: Go to Content Hosts -> Select content host -> Errata Tab ->
+        Select Previous environments (Environments Dropdown).
 
-    :expectedresults: The errata from previous environments are displayed.
+    :expectedresults: The previous environment name, and content view name are correct.
+        The errata dropdown from previous environments are displayed.
+
+    :bz: SAT-25213
 
     :parametrized: yes
     """
     vm = registered_contenthost
-    hostname = vm.hostname
+    nailgun_host = registered_contenthost.nailgun_host
     assert vm.execute(f'yum install -y {FAKE_1_CUSTOM_PACKAGE}').status == 0
     # Promote the latest content view version to a new lifecycle environment
-    content_view = module_cv.read()
     new_lce = module_target_sat.api.LifecycleEnvironment(
-        organization=module_org,
+        organization=module_sca_manifest_org,
         prior=module_lce,
     ).create()
-    content_view = content_view.read()
+    content_view = module_cv.read()
     content_view.version.sort(key=lambda version: version.id)
     content_view_version = content_view.version[-1].read()
-    content_view_version.promote(data={'environment_ids': new_lce.id})
+    content_view_version.promote(data={'environment_ids': [new_lce.id]})
+    # set host to use {new_lce / module_cv}, prior should be {module_lce / module_cv}
+    nailgun_host.content_facet_attributes = {
+        'lifecycle_environment_id': new_lce.id,
+        'content_view_id': module_cv.id,
+    }
+    nailgun_host.update(['content_facet_attributes'])
+    nailgun_host = nailgun_host.read()
+    # new_lce has been set for vm's Current Content Source
+    vm_cve = vm.nailgun_host.content_facet_attributes['content_view_environments'][0]
+    ng_cve = nailgun_host.content_facet_attributes['content_view_environments'][0]
+    assert vm_cve['lifecycle_environment'] == ng_cve['lifecycle_environment']
+    assert new_lce.name == vm_cve['lifecycle_environment']['name']
+    assert new_lce.id == vm_cve['lifecycle_environment']['id']
 
     with session:
         session.location.select(loc_name=DEFAULT_LOC)
         # can view errata from previous env, dropdown option is correct
         environment = f'Previous Lifecycle Environment ({module_lce.name}/{content_view.name})'
         content_host_erratum = session.contenthost.search_errata(
-            hostname,
+            vm.hostname,
```

Comment:
```suggestion
            registered_contenthost.hostname,
```

Reply:
Same thoughts as prior suggestion for Ln@1338